%YAML:1.0
---

# -----     华睿相机的标识字符串     -----
#Key: 'Huaray Technology:BA12121AAK00036'
# Key: 'Huaray Technology:3E04D1EAAK00038'
#Key: 'Dahua Technology:4H022F8PAK3B3A9'
Key: 'Dahua Technology:4H022F8PAK7FA33'
# -----     华睿相机的运行参数     -----
RuntimeParam:
  # 是否在线运行(布尔型)；在线运行：1，离线运行：0
  IsOnline: 1
  # 是否录像(布尔型)；录像：1，不录像：0
  IsRecordVideo: 1
  # 录像质量(整型)；取值范围：[1, 100]
  RecordVideoQuality: 50
  # 离线视频文件名称(字符串)
  OfflineVideoName: '/home/zhangtianyi/test_ws/src/cubot_radar/data/data1/video/2021-09-17/new2.mp4'
  # 录像文件存储路径(字符串)，必须以“/”结尾
  RecordVideoPath: '/home/zhangtianyi/test_ws/src/cubot_radar/data/data1/vide1/'

# -----     华睿相机的模型参数     -----
ModelParam:
  # 华睿相机OpenCv格式的内参矩阵
  CvInternalMatrix: !!opencv-matrix
    rows: 3
    cols: 3
    dt: d
    data: [1673.4446, 0,          652.3514,
           0,         1674.1376,  546.4549,
           0,         0,          1]

  # 华睿相机与Livox雷达的OpenCv格式的外参矩阵
  CvExternalMatrix: !!opencv-matrix
    rows: 4
    cols: 4
    dt: d
    data: [ 0.428848, -0.901283, 0.0614697, -1.65724,
            0.0143869,  -0.0612219, -0.99802, -0.17614,
            0.903262,    0.428884, -0.0132882, 0.396994,
            0,       0,        0,       1 ]
    #data: [-0.0325364, -0.999093,  0.0274572, 0.0981355,
    #       0.0763476,  -0.0298759, -0.996634, -0.269361,
    #       0.99655,    -0.0303305, 0.0772504, -0.894129,
    #       0,       0,        0,       1]

  # 华睿相机OpenCv格式的畸变校正向量·
  CvDistortionVector: !!opencv-matrix
    rows: 5
    cols: 1
    dt: d
    data: [-0.1549, 0.3642, 0.0007, -0.0008, 0.0212]
    #data: [-0.1549, 0.3642, 0.0007, -0.0008, 0.0212]

# -----     华睿相机的硬件参数     -----
HardwareParams:
  -
    # 相机硬件参数适配的场景；1-击打敌人，2-击打Buff
    Scene: 1
    # 帧率(浮点型)；取值范围：[1, 2000]
    FrameRate: 200.0
    # 是否自动曝光(布尔型)；自动曝光：1，非自动曝光：0
    IsExposureAuto: 0
    # 曝光时间(浮点型)；取值范围：
    ExposureTime: 7000.0
    # 增益值(浮点型)；取值范围：[1, 32]
    GainRaw: 4.0
    # 伽马值(浮点型)；取值范围：[0, 3.99998]
    Gamma: 0.65
    # 是否自动白平衡(布尔型)；自动白平衡：1，非自动白平衡：0
    IsWhiteBalanceAuto: 1
    # 红色平衡率(浮点型)；取值范围：[0, 15]
    BalanceRatio_R: 1.0
    # 绿色平衡率(浮点型)；取值范围：[0, 15]
    BalanceRatio_G: 1.0
    # 蓝色平衡率(浮点型)；取值范围：[0, 15]
    BalanceRatio_B: 1.0
    # 图像宽度(整型)；取值范围：[128, 1280]
    Width: 1280
    # 图像高度(整型)；取值范围：[64, 1024]
    Height: 1024
    # X轴偏移(整型)；取值范围：[0, 280]
    Offset_X: 0
    # Y轴偏移(整型)；取值范围：[0, 224]
    Offset_Y: 0
    # 相机亮度(整型)；取值范围：[0, 100]
    Brightness: 100
    # 选中状态（整型）
    IsSelected: 1

  -
    # 相机硬件参数适配的场景；1-击打敌人，2-击打Buff
    Scene: 2
    # 帧率(浮点型)；取值范围：[1, 2000]
    FrameRate: 200.0
    # 是否自动曝光(布尔型)；自动曝光：1，非自动曝光：0
    IsExposureAuto: 0
    # 曝光时间(浮点型)；取值范围：
    ExposureTime: 7000.0
    # 增益值(浮点型)；取值范围：[1, 32]
    GainRaw: 4
    # 伽马值(浮点型)；取值范围：[0, 3.99998]
    Gamma: 0.65
    # 是否自动白平衡(布尔型)；自动白平衡：1，非自动白平衡：0
    IsWhiteBalanceAuto: 1
    # 红色平衡率(浮点型)；取值范围：[0, 15]
    BalanceRatio_R: 1.0
    # 绿色平衡率(浮点型)；取值范围：[0, 15]
    BalanceRatio_G: 1.0
    # 蓝色平衡率(浮点型)；取值范围：[0, 15]
    BalanceRatio_B: 1.0
    # 图像宽度(整型)；取值范围：[128, 1280]
    Width: 1280
    # 图像高度(整型)；取值范围：[64, 1024]
    Height: 1024
    # X轴偏移(整型)；取值范围：[0, 280]
    Offset_X: 0
    # Y轴偏移(整型)；取值范围：[0, 224]
    Offset_Y: 0
    # 相机亮度(整型)；取值范围：[0, 100]
    Brightness: 100
    # 选中状态（整型）
    IsSelected: 0
